frontend/device/generic/powerd.lua

```diff
@@ -133,18 +150,45 @@ function BasePowerD:setIntensity(intensity)
 end
 
 function BasePowerD:getCapacity()
-    local now_ts = os.time()
-    if now_ts - self.last_capacity_pull_time >= 60 then
-        self.battCapacity = self:getCapacityHW()
+    -- NOTE: UIManager *should* be loaded at this point.
+    --       If that doesn't hold, c.f., :stateChanged below.
+    local UIManager = require("ui/uimanager")
+    local now_ts = UIManager:getTime()
+
+    if (now_ts - self.last_aux_capacity_pull_time):tonumber() >= 60 then
+        self.batt_capacity = self:getCapacityHW()
         self.last_capacity_pull_time = now_ts
```

Comment:
Shouldn't use `self.last_aux_capacity_pull_time` in if, but `self.last_capacity_pull_time` .

Reply:
Oops, nice catch, I should be able to fix it tonight, thanks!